content/blog/2020-07-31-spreadsheet-workflows-using-r/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+
+summary_table
+```
+
+## Row Calculations
+
+One thing that can be tricky when working with pivot tables is calculating new variables by rows as opposed to columns, with a usual work around being to manually add in a formula in a cell adjacent to our table.  Working with our data in R we can take advantage of the `rowwise()` function from dplyr, telling R we want to compute new values and summary statistics by row and not column.  
+Let's take a look at how much money was left over in each quarter by subtracting money spent from our budget and use the `rowwise()` function to specify we want this done by each row in our summary table
+```{r}
+left_over <- summary_table %>%
+  rowwise() %>%
+  mutate(left_over = total_budget_amt - total_expense_amt) %>%
+  arrange(left_over)
+
+left_over
+```
+Great! Using the `arrange()` function we can also bring the largest instances of overspending to the top of our table to focus our attention.
+
+We can also filter our table if we're interested in a specific expense type. Let's hone in on spending for office happy hours.
+```{r}
+left_over %>%
+  filter(expense == "office happy hour")
+```
+Hmm, a lot of spending in Quarter 3, which might make sense as these are the summer months. 
+
+## From tables to visualizations
+
+Finally, we can apply data visualsation to our workflow and look at spending across quarters and expense type. The ggplot2 package was loaded when we called the tidyverse but we'll add the scales package so we can put put a dollar format on our y axis of money left over. 
+```{r,message=FALSE}
+library(scales)
+left_over %>%
+  ggplot(aes(quarter, left_over)) +
+  geom_point() +
+  geom_line(group = 1) +
+  facet_wrap( ~ expense) +
+  scale_y_continuous(labels = dollar) + 
+  geom_hline(yintercept=20, 
+                color = "red") + 
+  labs (title  = "Spending by Expense Type",
+         x = "Quarter",
+         y = "$ Left Over")
+```
+Voila! - We have both a sumary table and a graphic that can tell us useful information about company spending. Hmm, there might be something to the trend we see with overexense on happy hours in Q3 after Q1 and Q2 were sunk with lawsuits
+
+
+## Join the Party
+Wherever you are in your spreadsheets journey, we've all had a V-Lookup formula mess up and spent more time than we'd like to admit trying to debug it.  V-Lookups are incredibly powerful and allow us to bring in outside data to further our analysis.  
+There are caveats however, the more lookups we perform we run the risk of slowing down our spreadsheet in addition to  issues of formula errors.  Using join functions from the dplyr package allows us to quickly bring in data from other tables in a simplified way and since we're working with our data in RStudio, lowers the chances of a significant slow down regardless of the amount of joins we conduct.  
+
+For this our next scenario, we'll leave the finance world and move to something perhaps a little more exciting, supheroes!
```

Comment:
This is the first spreadsheet I genuinely want to look at, ha! Superheroes are more fun that budgets.

What's the plan re: making them easy to see / download?

Reply:
Is that something we can add in the article itself? A way to download/view the whole data set I mean

Reply:
You can look at the flair post to see how to do this: https://github.com/rstudio/education.rstudio.com/blob/2b49964b0426bc7f0e00c129d1672cee2fc3f5b8/content/blog/2020-05-04-flair/index.Rmarkdown#L59-L66

Reply:
I think you either need to make it easy to download the Excel files you're using or, instead, work with world readable Google Sheets.

Either the way, the overall goal is that a motivated reader would be able to run the code in the post.

Reply:
I did it a little differently (by placing the sheets on a Google Drive), but I think that's ok.